alphagov/paas-release-ci
# [#157389026] Separate pipelines for tech-docs and rubbernecker

## What

Create individual pipelines for tech-docs and rubbernecker

We tried to handle the test/build/deploy process for all apps in a common pipeline but it resulted in multiple ifs and smaller hacks to handle all use cases. We decided that having individual pipelines for the applications will make modifications easier and result in simpler pipelines.

Also we moved all the deployment logic into the pipelines from the apps so we only have to modify the process in one place.

See [1] for more details.

[1] https://github.com/alphagov/paas-release-ci/pull/59/commits/8e991992da6506b8e3a45fda04b4d6067145cf1a

❗️ See the How to merge section as the branch contains multiple temporary commits.

## How to review

1. Deploy a CI dev environment from this branch

1. Review the following PRs while the new env is being deployed:

    * https://github.com/alphagov/paas-rubbernecker/pull/42
    * https://github.com/alphagov/paas-tech-docs/pull/137

1. Update the Concourse pipelines from this branch

    ```
    cd paas-release-ci
    BRANCH=separate_app_pipelines_157389026 SELF_UPDATE_PIPELINE=false CF_DEPLOY_ENV=<your CF env> DEPLOY_ENV=<your CI env> make dev pipelines
    ```

1. Upload the following secrets:

    ```
    cd paas-release-ci
    make upload-cf-cli-secrets
    make upload-rubbernecker-secrets
    ```

1. Run the setup pipeline

1. Enable the paas-tech-docs and rubbernecker projects and run them successfully

1. Go to https://paas-tech-docs.${DEPLOY_ENV}.dev.cloudpipelineapps.digital/, it should redirect you to https://docs.${DEPLOY_ENV}.dev.cloudpipeline.digital/

1. Check if https://docs.${DEPLOY_ENV}.dev.cloudpipeline.digital/ works correctly

1. Check if https://rubbernecker.${DEPLOY_ENV}.dev.cloudpipelineapps.digital/ works correctly

1. You should check whether all apps have the right environment variables.

## How to merge

1. Pause the paas-tech-docs and rubbernecker pipelines on the prod CI.

1. Merge the following PRs first:

    * https://github.com/alphagov/paas-rubbernecker/pull/42
    * https://github.com/alphagov/paas-tech-docs/pull/137

1. Wait until the new https://hub.docker.com/r/governmentpaas/tech-docs/ Docker image is built

1. Remove all temporary commits

1. Merge this PR and wait until the setup pipeline is finished on the prod CI.

1. Reenable the paas-tech-docs and rubbernecker pipelines on the prod CI.

1. Check if the pipelines run successfully.

## Who can review

Not me.
